### DISCUSSION TITLE
Tesla-History Tool for InfluxDB2+?

### DISCUSSION THREAD

**Discussion by ThePnuts** [jasonacox/Powerwall-Dashboard#384] [Q&A]
I dont suppose there happens to be a version of this tool that works with InfluxDB2+ ?

It seems I didnt log the last 7 days of data, still need to figure out why, but now I need to get that data in there too.

**Comment by ThePnuts** [comment]
I manually converted downloaded Tesla data and imported it for now.

My issue was caused by the time change and Telegraf locking up.

Just realized I am running influx2 in parallel to 1.8 since I was comparing the data from what I did in Flux to the official data. That Telegraf agent did not lock up, so I have all of the data there, just need to figure out how to move it over.

I looked at the script to automate the Tesla data, its a good bit over my head so that might not be something I try to tackle.

**Reply by jasonacox** [reply]
Hi @ThePnuts, if you figure out a good script to do the data migration, would you mind sharing it?  At some point we may be pivoting the project from 1.8 to 2 so we will need a migration tool. 🙏 

Did you figure out why the influx2 telegraf locked out? It seems odd that a time change would do that.  But let me guess, you are using a bleeding edge version of telegraf? 😉

**Reply by ThePnuts** [reply]
> Did you figure out why the influx2 telegraf locked out? It seems odd that a time change would do that. But let me guess, you are using a bleeding edge version of telegraf? 😉

You know I am, lol
I honestly don't know. There was a pending docker update for it, so I did that as a step 1 to solving my problem and it did. I only figured out it was the time change when manually looking to add the tesla downloaded data back in and realized it stopped there. If I get time, ill check if there is a not on what the update to telegraf did after the time change.

Whats funny is that I am using the same latest version of telegraf docker on the 1.8 influxdb. That docker kept running, go figure.


> Hi @ThePnuts, if you figure out a good script to do the data migration, would you mind sharing it? At some point we may be pivoting the project from 1.8 to 2 so we will need a migration tool. 🙏

Its on my list now so I can get that data I lost, dunno when I'll get to it though. I'll definitely share it.

I also have on my list to kind of re-do the Grafana page I did in Flux as well. I made a few others since then and learned functionality to kind of auto scale for sample size, so I can increase the time range without it dropping data in the graphs. Like selecting the last year only has data for 8 months in the graph due to limits.

**Comment by mcbirse** [comment]
Hi @ThePnuts - I looked at what would be involved in trying to add InfluxDB2+ support to the tesla-history script a while ago, and it appeared to be non-trivial. 😞 

From memory the main issue was around the InfluxDB queries and query updates that update the analysis data, rather than necessarily writing data to InfluxDB as such...

Since there didn't seem to be any demand for InfluxDB2+ support at this stage, I put it in the too hard basket for now. I may look at it again in the future, but can't really help with your immediate issue until I revisit this.

**Reply by ThePnuts** [reply]
Haha, yeah, I'm not sure how many people are using InfluxDB2+, I would bet less then 5, since this project is pretty turnkey to setup as-is without issue.

I am able to do it semi manually. I just DL the days needed from the Tesla App and let excel do the heavy work to convert it to a template I made for import. Then run 2 flux queries after adjusting the time window and I am back in business for what Tesla provides.

[Tesla Data Manual Import-Template.csv](https://github.com/jasonacox/Powerwall-Dashboard/files/13369096/Tesla.Data.Manual.Import-Template.csv)


First
```
//tesla_energy.instant_power <= raw_tesla_export
//Change start date to start of missing data from raw export data
//Change stop date to end of missing data

from(bucket: "raw_tesla_export")
  |> range(start: 2023-11-05T09:00:00.000Z, stop: 2023-11-13T04:34:00.000Z)
  |> filter(fn: (r) => r["_measurement"] == "teslaexport")
  |> map(fn: (r) => ({r with _value: r._value * 1000.0}))
  |> map(fn: (r) => ({r with _measurement: "instant_power"}))
  |> drop(columns: ["_start", "_stop"])
  |> to(bucket: "tesla_energy")
```

2nd for kWh calculations

```
//tesla_energy.kwh <= tesla_energy.instant_power
//Change start date to start of missing data from raw export data

import "math"
import "date"

from(bucket: "tesla_energy")
  |> range(start: date.truncate(t: 2023-11-05T09:00:00.000Z, unit: 1h), stop: date.truncate(t: now(), unit: 1h))
  |> filter(fn: (r) => r["_measurement"] == "instant_power")
  |> filter(fn: (r) => r._field == "home" or r._field == "solar" or r._field == "grid" or r._field == "to_grid" or r._field == "from_grid" or r._field == "powerwall" or r._field == "to_powerwall" or r._field == "from_powerwall")
  |> aggregateWindow(every: 1h, fn: integral, createEmpty: false)
  |> map(fn: (r) => ({r with _time: date.sub(d: 1h, from: r._time)}))
  |> map(fn: (r) => ({r with _measurement: "kwh"}))
  |> map(fn: (r) => ({r with _value: r._value / 1000.0 / 3600.0}))
  |> to(bucket: "tesla_energy")
```
